feat: implement API rate limiting

baiyaoshi opened the pull request.

baiyaoshi committed:
feat: implement interface rate limit with sliding window

baiyaoshi committed:
feat: implement interface rate limit with sliding window

baiyaoshi committed:
fix:fix some test issues

RooobinYe commented:
写的很好，但是有那么几点问题，是我当时没有和你说清楚
1. 我认为当前的代码侵入性太强，我更加倾向的实现是使用 AOP，在需要限流的地方添加一个注解
2. `public boolean tryAcquire` 这个方法的实现是有点问题的，建议使用 Sorted Set or ZSet [Redis sorted sets | Docs](https://redis.io/docs/latest/develop/data-types/sorted-sets/)
3. 此外，当初的方案没法保证操作的原子性，比如：如果有大量的请求发送过来，可能会有两个请求，申请操作同一个 Redis 的字段，推荐使用 Bucket4j 这个第三方库，参考的文章是 [Implementing Rate Limiting with Redis and Spring Boot: A Deep Dive. | by Prajwal Patil | Medium](https://medium.com/@prajpatil29/implementing-rate-limiting-with-redis-and-spring-boot-a-deep-dive-0bd000bcd98c) 考虑过直接使用 Lua 脚本，但是不如一步到位了。「之前我说的有点问题，这个功能实现使用令牌桶算法吧，而不是滑动窗口。」
4. 其他我想说的内容：限流的容量和时间直接写死吧，我觉得问题不是很大；单独为 user 设置限流就行了，单独文件的限流意义也不是很大，总不能一个人下载完了，不让别人下载吧
有任何的问题和我讲，一起讨论

cc (Carbon Copy) @Hunger61

baiyaoshi committed:
feat: implement rate limiting using AOP

baiyaoshi committed:
fix: fix JwtUtilTest issues

RooobinYe committed:
refactor: optimize Redis configuration management

- Move Redis configuration from @PostConstruct to dedicated @Configuration class
- Create RedisConfig in approval-server/config package for better separation of concerns
- Remove redundant serializer setup from RedisUtil utility class
- Improve code organization following Spring Boot best practices

RooobinYe reviewed (approved):
LGTM 🎉

RooobinYe commented:
@Hunger61 PTLA
